contracts/libraries/UniERC20.sol

```diff
@@ -106,6 +131,38 @@ library UniERC20 {
         }
     }
 
+    /**
+     * @dev Transfers a specified amount of the token from one address to another.
+     * Note: Does nothing if the amount is zero.
+     * Note 2: Uses the SafeSendLib for safe Ether transfers.
+     * @param token The token to transfer.
+     * @param from The address to transfer the token from.
+     * @param to The address to transfer the token to.
+     * @param amount The amount of the token to transfer.
+     */
+    function uniSafeTransferFrom(
+        IERC20 token,
+        address payable from,
+        address to,
+        uint256 amount
+    ) internal {
+        if (amount > 0) {
+            if (isETH(token)) {
+                if (msg.value < amount) revert NotEnoughValue();
+                if (from != msg.sender) revert FromIsNotSender();
+                if (to != address(this)) revert ToIsNotThis();
+                if (msg.value > amount) {
+                    // Return remainder if exist
+                    unchecked {
+                        from.safeSend(msg.value - amount);
```

Comment:
The method sends ETH from `msg.sender` to somewhere vie eip7702? It's not obvious where the funds will go despite that the description is they have to go to the `to`